### PR TITLE
Role Rework: Mayor -> Politician

### PR DESCRIPTION
**Pull request by taelspinner**
Mayor is a somewhat troubled role in Town of Us. Originally, it worked by giving the Mayor an extra vote each meeting, which they could save, stockpiling votes for a crucial moment later in the game. Later, it was reworked for balance reasons to be a "guaranteed innocent" role, which can reveal itself at that crucial moment to gain a fixed number of extra votes and immunity from suspicion, at the cost of being very vulnerable and a priority target in subsequent rounds.

This still leaves Mayor as a rather lackluster role for crew. The Mayor has nothing to really _do_, aside from try to stay alive long enough to tip the balance of the match. The ability to reveal at any time is also somewhat suboptimal, even if the Mayor is discouraged from using it, because it allows someone to clear themselves at the very first meeting with the press of a button if so inclined (which also pretty much guarantees the Mayor will not survive to have a significant effect on the voting). It's boring, and probably the second-least fun role behind Detective (and maybe Vigilante).

This PR reworks Mayor to function as a town-aligned Plaguebearer. A new role, called the Politician, has been added which the Mayor will begin as. While a Politician, the player is as vulnerable as any other crewmate and has no ability to clear themselves. They gain an active ability, Campaign, which functions identically to Infect, gaining the vote (for Mayor) of the affected player, and potentially putting themselves at risk for Veteran and Pestilence. Like infected players, campaigned players spread with all interactions. This also makes Politician a potentially useful intelligence-gathering role, as they can glean some information about who has interacted with whom. (This is in some cases very powerful, and therefore there is a toggle, defaulting off, for whether the Politician can see infected players in the meetings.)

When all players have been campaigned, the Politician transforms into the Mayor. The Mayor no longer has a vision radius penalty since they now have to put in the work to become the Mayor, and additionally gains an active ability, Bodyguard. The Bodyguard ability functions identically to Veteran's Alert, and resets to 1 use after each meeting, beginning with 0. At the next meeting, the Mayor will be automatically revealed to all living players, but as they can only reveal during meetings, during the round in which they become the Mayor, they are _not_ revealed.

In summary:
- Mayor is now to Politician what Pestilence is to Plaguebearer.
- The Politician Campaigns, functioning very similarly to Infect.
- When all living players are Campaigned, the Politician transforms into the Mayor.
- The Mayor no longer has a vision penalty.
- The Mayor now automatically reveals at the next meeting after they become the Mayor.
- The Mayor has a Bodyguard ability that functions very similarly to Alert to protect them after they become the Mayor.
- Because the Mayor has a Veteran-like ability, the Politician has been moved to the Crewmate Killing category alongside the Veteran.

All code and assets are original.

### CODE REVIEW

**Comment by fariparedes** [comment]
Updated for v5.0.4
@MyDragonBreath Thoughts on this change?

**Comment by eDonnes124** [comment]
I took the idea and went in a different direction...
